src/lib/esnext.intl.d.ts

```diff
@@ -1,3 +1,168 @@
+     */
+    type fractionalDigitsOption = 0|1|2|3|4|5|6|7|8|9
+
+    interface ResolvedDurationFormatOptions {
+        locale?: UnicodeBCP47LocaleIdentifier
+        numberingSystem?: DateTimeFormatOptions["numberingSystem"]
+        style?: DurationFormatStyle
+        years?: Omit<DurationFormatOptions,"numeric"|"2-digit">
+        yearsDisplay?:DurationFormatDisplayOption
+        months?: Omit<DurationFormatOptions,"numeric"|"2-digit">
+        monthsDisplay?: DurationFormatDisplayOption
+        weeks?: Omit<DurationFormatOptions,"numeric"|"2-digit">
+        weeksDisplay?: DurationFormatDisplayOption
+        days?: Omit<DurationFormatOptions,"numeric"|"2-digit">
+        daysDisplay?: DurationFormatDisplayOption
+        hours?: DurationFormatOptions
+        hoursDisplay?: DurationFormatDisplayOption
+        minutes?: DurationFormatOptions
+        minutesDisplay?: DurationFormatDisplayOption
+        seconds?: DurationFormatOptions
+        secondsDisplay?: DurationFormatDisplayOption
+        milliseconds?: DurationFormatOptions
+        millisecondsDisplay?: DurationFormatDisplayOption
+        microseconds?: DurationFormatOptions
+        microsecondsDisplay?: DurationFormatDisplayOption
+        nanosecond?: DurationFormatOptions
+        nanosecondDisplay?: DurationFormatDisplayOption
+        fractionalDigits?: fractionalDigitsOption
+    }
+
+    interface DurationFormatOptions {
+        localeMatcher?: DurationTimeFormatLocaleMatcher
+        numberingSystem?: DateTimeFormatOptions["numberingSystem"]
+        style?: DurationFormatStyle
+        years?: Omit<DurationFormatOption,"numeric"|"2-digit">
+        yearsDisplay?:DurationFormatDisplayOption
+        months?: Omit<DurationFormatOption,"numeric"|"2-digit">
+        monthsDisplay?: DurationFormatDisplayOption
+        weeks?: Omit<DurationFormatOption,"numeric"|"2-digit">
+        weeksDisplay?: DurationFormatDisplayOption
+        days?: Omit<DurationFormatOption,"numeric"|"2-digit">
+        daysDisplay?: DurationFormatDisplayOption
+        hours?: DurationFormatOption
+        hoursDisplay?: DurationFormatDisplayOption
+        minutes?: DurationFormatOption
+        minutesDisplay?: DurationFormatDisplayOption
+        seconds?: DurationFormatOption
+        secondsDisplay?: DurationFormatDisplayOption
+        milliseconds?: DurationFormatOption
+        millisecondsDisplay?: DurationFormatDisplayOption
+        microseconds?: DurationFormatOption
+        microsecondsDisplay?: DurationFormatDisplayOption
+        nanosecond?: DurationFormatOption
+        nanosecondDisplay?: DurationFormatDisplayOption
+        fractionalDigits?: fractionalDigitsOption
+    }
+
+    /**
+     * The duration object to be formatted
+     * 
+     * [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DurationFormat/format#duration).
+     */
+    type DurationType = Record<DurationTimeFormatUnit,number>

```

Comment:
tried this locally but with strict type settings this causes all fields to be required. I believe it is valid for the type to be partial, like `{hours: 1, minutes: 30}` so this seems like it should be:

```suggestion
    type DurationType = Partial<Record<DurationTimeFormatUnit,number>>

```